src/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.cpp

```diff
@@ -120,6 +121,17 @@ void Propagator::setDynamics(const Array<Shared<Dynamics>>& aDynamicsArray)
     }
 }
 
+void Propagator::addManeuvers(const Array<Maneuver>& aManeuverArray)
+{
+    for (const Maneuver& aManeuver : aManeuverArray)
+    {
+        const Shared<TabulatedDynamics> tabulatedDynamics =
+            aManeuver.toTabulatedDynamics(IntegrationFrameSPtr, aManeuver.getInterpolationType());
+
+        this->addDynamics(std::dynamic_pointer_cast<Dynamics>(tabulatedDynamics));
```

Comment:
```suggestion
        this->addDynamics(tabulatedDynamics);
```
Don't think the dynamic pointer cast is necessary.

Reply:
oh yeah I guess you're right, you only need to do that when you want to dereference the object afterwords right?